# Planner selection
base_global_planner: "navfn/NavfnROS"
base_local_planner: "base_local_planner/TrajectoryPlannerROS"

# Recovery behaviors are defined in robot folders

# Oscillation
oscillation_timeout: 10.0
oscillation_distance: 0.5

# Global planner
planner_frequency: 0.5 #0.1
#shutdown_costmap: true
planner_patience: 5.0
NavfnROS:
  # allow_unknown: false  # investigate this
  allow_unknown: true  # investigate this
  default_tolerance: 0.0
  # do not restrict planner
  planner_window_x: 0.0
  planner_window_y: 0.0
  # debug
  visualize_potential: false

# Local planner
controller_frequency: 10.0
controller_patience: 5.0
max_planning_retries: 5
TrajectoryPlannerROS:
  # base vel/accel profile is in robot folders
  # tolerances (defaults)
  yaw_goal_tolerance: 0.05 #0.1
  xy_goal_tolerance: 0.1 #0.1
  latch_xy_goal_tolerance: true
  # forward simulation
  sim_time: 2.5
  sim_granularity: 0.025
  angular_sim_granularity: 0.025
  vx_samples: 3
  vtheta_samples: 10 # scoring (defaults) meter_scoring: false #true 
  pdist_scale: 1.4 #0.8 #0.5 #6.0
  gdist_scale: 1.0 #1.2 #0.4 #6.0
  occdist_scale: 0.1 # 0.06 #0.6
  heading_lookahead: 3.0 #0.35
  heading_scoring_timestep: 0.8
  heading_scoring: false
  dwa: false
  # other
  oscillation_reset_dist: 0.05
  # debug
  publish_cost_grid_pc: false
